chapter05/red/exercise2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use List::Util qw( reduce );
```

Comment:
Fold is as fundamental as map and filter (`reduce`, `map`, and `grep` in perl, respectively). Many languages teach them as one concept ([e.g](http://www.cse.unsw.edu.au/~en1000/haskell/hof.html)). Totally not cheating to use this.